Changelog for mlflow 1.7 (#2520)
mlflow/mlflow: CHANGELOG.rst

Changelog
=========

> 1.7.0 (2020-03-01)

Reviewer: @mparkhe I don't see entries here for the `mlflow gc` command. Were cherry-picked features not included in this autogenerated change log?

cc @smurching

Reviewer 2: Ah, we need to add labels to PRs between the 1.6 & 1.7 releases for them to show up in the changelog (will update the automation to throw when there are unlabeled PRs, & also go back and label the old PRs)

Reviewer 2: Ideally we could also enforce that a PR has a release-notes label before merging

Reviewer 3: yeah! i added labels to most of the PRs -- i must have forgotten the original gc PR

------------------
MLflow 1.7.0 includes several major features and improvements:

> Model Registry APIs: breaking changes

Reviewer 4: This is only REST APIs, right, not Python APIs? Most users won't see the REST ones at all so it would be good to say REST APIs. Also say what the impact is (which clients are compatible with new/old servers: do you have to upgrade both? if not, how long is compatibility going to be supported for?)

Reviewer 3: Thanks @mateiz. Made the title more clear and added comments about clients maintaining backward compatibility at the API level and the user code does not need to change.

> - REST APIs are no longer in beta and intended to be stable until the next major version. As such, this release includes a number of breaking changes.

Reviewer 4: It would be nice to mention why they changed, i.e. to clean up the API or make it more consistent with other APIs.

- Model Registry UI has been updated to use new REST APIs (#2456, #2476 @aarondav; #2507, @mparkhe)
- MLflow Client APIs are backward compatible and internally been updated to use new REST APIs (#2457, @tomasatdatabricks; #2502, @mparkhe)
- Java Client API to download models artifacts (#2308, @andychow-db)
- Updated Model Registry documentation page with code snippets and examples (#2493, dmatrix)

Other Features:

> - Fix bar chart bug introduced in #2393 (#2498, @smurching)

Reviewer 4: People don't know what #2393 is so say which MLflow release this was in (or even better, which bug)

- Save metric plot state in URLs for easier link sharing (#2393, @smurching)

> - Update R client to call record logged model  (#2430, @tomasatdatabricks)

Reviewer 4: Isn't this part of the "add log model endpoint"? We should group these into topics, instead of having a separate line for each PR.

- Fix handleLayoutChange to update dragmode (#2459, @harupy)

> - Announce deprecation of Python 2 support in MLflow (#2438, @smurching)

Reviewer 4: This should say "this release deprecates Python 2", and it's major enough that it should be part of the major changes.

Changelog
=========
1.7.0 (2020-03-02)
------------------
MLflow 1.7.0 includes several major features and improvements, and some notable breaking changes:

MLflow support for Python 2 is now deprecated and will be dropped in a future release. At that
point, existing Python 2 workflows that use MLflow will continue to work without modification, but
Python 2 users will no longer get access to the latest MLflow features and bugfixes. We recommend
that you upgrade to Python 3 - see  https://docs.python.org/3/howto/pyporting.html for a migration
guide.

> Breaking changes to Model Registry REST APIs:

Reviewer 5: i don't think we need this heading, but let's check this in for the release now - we can always tweak later.

Changelog
=========
1.7.0 (2020-03-02)
------------------
MLflow 1.7.0 includes several major features and improvements, and some notable breaking changes:

MLflow support for Python 2 is now deprecated and will be dropped in a future release. At that
point, existing Python 2 workflows that use MLflow will continue to work without modification, but
Python 2 users will no longer get access to the latest MLflow features and bugfixes. We recommend
that you upgrade to Python 3 - see  https://docs.python.org/3/howto/pyporting.html for a migration
guide.

Breaking changes to Model Registry REST APIs:

> Model Registry REST APIs are updated to be more consistent with the other MLflow APIs. With this

Reviewer 6: ```suggestion
Model Registry REST APIs have been updated to be more consistent with the other MLflow APIs. With this
```

release Model Registry APIs are no longer beta and intended to be stable until the next major
version. Python and Java client APIs for Model Registry are backward compatible and have been
updated to use new REST APIs. However, when using MLflow client with a server using updated REST

> endpoints, you will need to upgrade to new client version without changing any user code.

Reviewer 5: also, we can refer to the relevant PRs here instead of listing them in "Other Features" (now they are doubly mentioned)

> endpoints, you will need to upgrade to new client version without changing any user code.

Reviewer 6: However, when using MLflow client with a server using updated REST
endpoints, you will need to upgrade to new client version without changing any user code.

-> However, when using an MLflow client with a server using updated REST
endpoints, you won't need to change any code but will need to upgrade to a new client version.

> endpoints, you will need to upgrade to new client version without changing any user code.

Reviewer 3: I will move the API related stuff (UI and client you marked below)

Other Features:

> - Adds ability to click through to individual runs from metrics plot (#2295, @harupy)

Reviewer 6: ```suggestion
- Added ability to click through to individual runs from metrics plot (#2295, @harupy)
```

Reviewer 6: Change Adds to Added throughout.

- Adds ``mlflow gc`` CLI for permanent deletion of runs (#2265, @t-henri)
- Metric plot state is now captured in page URLs for easier link sharing (#2393, #2408, #2498 @smurching; #2459, @harupy)
- Adds experiment management to MLflow UI (create/rename/delete experiments) (#2348, @ggliem)
- Adds ability to search for experiments by name in the UI (#2324, @ggliem)
- MLflow UI page titles now reflect the content displayed on the page (#2420, @AveshCSingh)
- Adds a new ``LogModel`` REST API endpoint for capturing model metadata, and call it from the Python and R clients (#2369, #2430, #2468 @tomasatdatabricks)
- The Model Registry UI has been updated to use new REST APIs (#2476 @aarondav; #2507, @mparkhe)

> - Updated MLflow Python and Java clients to use new REST APIs. The client APIs have minor changes and deprecated arguments however for this release are backward compatible. These deprecated arguments will be dropped in future releases. (#2457, @tomasatdatabricks; #2502, @mparkhe)

Reviewer 5: merge these into the breaking changes blurb above

- Java Client API to download model artifacts (#2308, @andychow-db)

Bug fixes and documentation updates:

- Updated Model Registry documentation page with code snippets and examples (#2493, dmatrix)
- Better error message for Model Registry, when using incompatible backend server (#2456, @aarondav)
- matplotlib is no longer required to use XGBoost and LightGBM autologging (#2423, @harupy)

> - Fixes bug where matplotlib figures were not closed in XGBoost and LightGBM autologging (#2386, @harupy)

Reviewer 6: ```suggestion
- Fix bug where matplotlib figures were not closed in XGBoost and LightGBM autologging (#2386, @harupy)
```

> - Fix parameter reading logic to support param values with newlines in FileStore (#2376, @dbczumar)

Reviewer 6: ```suggestion
- Fixed parameter reading logic to support param values with newlines in FileStore (#2376, @dbczumar)
```

- Improve readability of run table column selector nodes (#2388, @dbczumar)
- Validate experiment name supplied to ``UpdateExperiment`` REST API endpoint (#2357, @ggliem)

> - Fix broken MLflow DB README link in CLI docs (#2377, @dbczumar)

Reviewer 6: ```suggestion
- Fixed broken MLflow DB README link in CLI docs (#2377, @dbczumar)
```

> - Change Copyright year across docs to 2020 (#2349, @ParseThis)

Reviewer 6: ```suggestion
- Change copyright year across docs to 2020 (#2349, @ParseThis)
```

Breaking changes to Model Registry REST APIs:

Model Registry REST APIs have been updated to be more consistent with the other MLflow APIs. With this
release Model Registry APIs are intended to be stable until the next major version. Python and Java
client APIs for Model Registry are backward compatible and have been updated to use the new REST
APIs. However, when using an MLflow client with a server using updated REST endpoints, you won't
need to change any code but will need to upgrade to a new client version.
- The Model Registry UI has been updated to use the new REST APIs (#2476 @aarondav; #2507, @mparkhe)
- Updated MLflow Python and Java clients to use the new REST APIs. The client APIs have minor changes and deprecated arguments however for this release are backward compatible. These deprecated arguments will be dropped in future releases. (#2457, @tomasatdatabricks; #2502, @mparkhe)

Other Features:

- Added ability to click through to individual runs from metrics plot (#2295, @harupy)

> - Adds ``mlflow gc`` CLI for permanent deletion of runs (#2265, @t-henri)

Reviewer 5: "Added" here and below
